controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -315,6 +315,13 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 		machine.Spec.NodeDeletionTimeout = kcp.Spec.MachineTemplate.NodeDeletionTimeout
 	}
 
+	// In case this machine is being created as a consequence of a remediation, then add an annotation
+	// tracking the name of the machine we are remediating for.
+	// NOTE: This is required in order to track remediation retries.
+	if v, ok := kcp.Annotations[controlplanev1.RemediationInProgressAnnotation]; ok && v == "true" {
+		machine.Annotations[controlplanev1.RemediationForAnnotation] = kcp.Status.LastRemediation.Machine
+	}
```

Comment:
Instead of this being an annotation on KCP (following up on the above comment about tracking state on Machines) could we look at the current state of things and determine if remediation is in progress or not?

Reply:
I fear not.
After the first part of remediation, the failing machine is deleted, and reconcile loop completes with requeue=true.

When the next reconcile happens, as soon as the rollout policy allows it, the reconciler enters the scale-up logic to restore the desired state of replicas. However, without this annotation, at this stage, we do not know if the machine has been deleted due to reconciliation or due to any other reason.

Reply:
Done! Now last remediation is tracked on machines and KCP rebuilds it status using this as a source of trouth